src/Planners/src/QuinticSpline.cpp

```diff
@@ -0,0 +1,787 @@
+}
+
+void QuinticSpline::Impl::addKnownTermPreviousKnot(const std::size_t& knotIndex,
+                                                   const std::size_t& coordinateIndex,
+                                                   Eigen::Ref<Eigen::VectorXd> b)
+{
+    const auto& poly = polynomials;
+
+    const auto& i = knotIndex;
+    const auto& j = coordinateIndex;
+
+    Eigen::Matrix2d tempMatrix;
+    tempMatrix << 8 / std::pow(poly[i - 1].duration, 2), 1 / poly[i - 1].duration,
+                 14 / std::pow(poly[i - 1].duration, 3), 2 / std::pow(poly[i - 1].duration, 2);
+
+    Eigen::Vector2d tempVector;
+    tempVector << knots[i - 1].velocity[j], knots[i - 1].acceleration[j];
+    b -= tempMatrix * tempVector;
+}
+
+void QuinticSpline::Impl::setPolynomialCoefficients(Polynomial& poly)
+{
+    const auto& T = poly.duration;
+
+    const auto& x0 = poly.initialPoint->position;
+    const auto& dx0 = poly.initialPoint->velocity;
+    const auto& ddx0 = poly.initialPoint->acceleration;
+
+    const auto& xT = poly.finalPoint->position;
+    const auto& dxT = poly.finalPoint->velocity;
+    const auto& ddxT = poly.finalPoint->acceleration;
+
+    poly.a0 = x0;
+    poly.a1 = dx0;
+    poly.a2 = ddx0 / 2.0;
+    poly.a3 = 1.0 / (T * T * T)
+              * (x0 * 2.0E+1 - xT * 2.0E+1 + T * dx0 * 1.2E+1 + T * dxT * 8.0 + (T * T) * ddx0 * 3.0
+                 - (T * T) * ddxT)
+              * (-1.0 / 2.0);
+    poly.a4 = (1.0 / (T * T * T * T)
+               * (x0 * 3.0E+1 - xT * 3.0E+1 + T * dx0 * 1.6E+1 + T * dxT * 1.4E+1
+                  + (T * T) * ddx0 * 3.0 - (T * T) * ddxT * 2.0))
+              / 2.0;
+    poly.a5 = 1.0 / (T * T * T * T * T)
+              * (x0 * 1.2E+1 - xT * 1.2E+1 + T * dx0 * 6.0 + T * dxT * 6.0 + (T * T) * ddx0
+                 - (T * T) * ddxT)
+              * (-1.0 / 2.0);
+}
+
+void QuinticSpline::Impl::computeIntermediateVelocitiesAndAccelaration()
+{
+    // here we assume that at least 3 points has been defined
+    const std::size_t numberOfInteriorKnots = knots.size() - 2;
+
+    Eigen::SparseMatrix<double> A(2 * numberOfInteriorKnots, 2 * numberOfInteriorKnots);
+    std::vector<Eigen::Triplet<double>> tripletsList;
+
+    std::size_t numberOfExpectedTriplets = 4;
+    if(numberOfInteriorKnots  > 1)
+    {
+        numberOfExpectedTriplets = 2 * (6 * (numberOfInteriorKnots - 2) + 2 * 4);
+    }
+    tripletsList.reserve(numberOfExpectedTriplets);
+
+    for (int i = 0; i < numberOfInteriorKnots; i++)
+    {
+        const int absoluteKnotIndex = i + 1;
+
+        // in this case there is only one interior knot A will be a 2
+        if (i == 0 && (i + 1) == numberOfInteriorKnots)
+        {
+            this->addTripletCurrentKnot(absoluteKnotIndex, 0, 0, tripletsList);
+        }
+        else if (i == 0)
+        {
+            this->addTripletCurrentKnot(absoluteKnotIndex, i * 2, 0, tripletsList);
+            this->addTripletNextKnot(absoluteKnotIndex, i * 2, 2, tripletsList);
+
```

Comment:
this extra line could be removed.

Reply:
Done in 97ebb0bfe8ea740c1318b2601865ae869e0ca41c

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-framework/pull/83/commits/97ebb0bfe8ea740c1318b2601865ae869e0ca41c